pong.h

```diff
@@ -12,12 +12,36 @@ namespace pong
 class Pong
 {
 private:
-    std::vector<std::unique_ptr<GameObject>>mGameObjects {};
+    Pong() = default;
+    Pong(const Pong&) = delete;
+    Pong& operator=(const Pong&) = delete;
+    Pong(Pong&&) = delete;
+    Pong& operator=(Pong&&) = delete;
+    ~Pong() = default;
+
+    std::vector<std::shared_ptr<GameObject>>mGameObjects {};
```

Comment:
Very curious what you think of this. I changed this to a shared_ptr array instead of unique_ptr. Realisitically I think GameObjects really are only useful when they are being using in the game itself. i.e. actually exits in this vector and have Update() and Render() called on them. So this class is generally going to be the sole "owner" of them. Other gameobjects can certainly have references to gameobjects. But, I want something similar to Unity's Destroy() api. This removes a gameobject during runtime. After this is called, the GameObject should probably not be used anymore, so I am curious if shared_ptr is better than unique_ptr?

Secondary question, if I do use unique_ptr, if another gameobject needs a reference to the same gameobject it would likely be using a reference&? I figured raw pointers are avoided, but unique_ptr could also not be used for the type, because that would cause memory to be released twice?

Reply:
> So this class is generally going to be the sole "owner" of them.

This is the strongest case for it to be unique pointer

> Other gameobjects can certainly have references to gameobjects.

To hand out a non-owning handle to an object, give out a raw pointer or raw reference to it with

```
doSomething(myPointer.get())

doSomethingRef(*myPointer.get()); // Note this could copy it if a calling function doesn't pass by reference
```

Reply:
Yep, makes sense, changed it back to unique_ptr